lib/omnibus/software.rb

```diff
@@ -773,15 +773,26 @@ def with_standard_compiler_flags(env = {}, opts = {})
         extra_linker_flags["LD_OPTIONS"] = ld_options
       end
 
-      env.merge(compiler_flags).
-        merge(extra_linker_flags).
-        # always want to favor pkg-config from embedded location to not hose
-        # configure scripts which try to be too clever and ignore our explicit
-        # CFLAGS and LDFLAGS in favor of pkg-config info
-        merge({ "PKG_CONFIG_PATH" => "#{install_dir}/embedded/lib/pkgconfig" }).
-        # Set default values for CXXFLAGS and CPPFLAGS.
-        merge("CXXFLAGS" => compiler_flags["CFLAGS"]).
-        merge("CPPFLAGS" => compiler_flags["CFLAGS"])
+      env = env.merge(compiler_flags).
```

Comment:
As I stare at this, I think the .merge.merge.merge.merge feels weird. Experimenting what about something like

```
hashes = [
  {a: 1},
  {a: 'a', b: 2},
  {c: 3},
]

hashes.reduce(:merge)
```

And the various choices are about which arrays to wedge into hashes. Kinda depends on how many hashes you actually have to merge. Some of these seem like they can be replaced and done in one shot.